smart-contracts/contracts/CardPurchaser.sol

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import "./interfaces/IUSDC.sol";
+import "./interfaces/IPublicLock.sol";
+import "./interfaces/IUnlock.sol";
+
+/// @custom:security-contact hello@unlock-protocol.com
+contract CardPurchaser is Ownable, EIP712 {
+  error WITHDRAW_FAILED();
+  error MISSING_LOCK();
+  error LOCK_CALL_FAILED();
+  error TOO_MUCH_SPENT();
+  error TOO_LATE();
+  error PURCHASER_DOES_NOT_MATCH_PAYER();
+  error SIGNER_DOES_NOT_MATCH();
+
+  // Unlock address on current chain
+  address public unlockAddress;
+
+  // Address of USDC contract
+  address public usdc;
+
+  // Name of contract (OZ's EIP712 does not expose it)
+  string public name;
+
+  // VErsion of contract (OZ's EIP712 does not expose it)
+  string public version;
+
+  struct ApprovalMessage {
+    address from;
+    address to;
+    uint256 value;
+    uint256 validAfter;
+    uint256 validBefore;
+    bytes32 nonce;
+  }
+
+  struct PurchaseMessage {
+    address lock;
+    address sender;
+    uint256 expiration;
+  }
+
+  /**
+   * Constructor
+   * We double the `name` and `version` fields from EIP712 so we can query the contract to get them and OZ's Ownable does not expose them
+   */
+  constructor(
+    address _owner,
+    address _unlockAddress,
+    address _usdc
+  ) EIP712("Card Purchaser", "1") Ownable() {
+    name = "Card Purchaser";
```

Comment:
You should be able to access is using `super.name` no?

Reply:
Let me try but `name` is a `private` var in `EIP712` :(